extensions/ql-vscode/package.json

```diff
@@ -346,6 +346,12 @@
           "default": "",
           "patternErrorMessage": "Please enter a valid folder",
           "markdownDescription": "The name of the folder where we want to create queries and query packs via the \"CodeQL: Create Query\" command. The folder should exist."
+        },
+        "codeQL.autogenerateQlPacks": {
+          "type": "string",
+          "default": "",
+          "patternErrorMessage": "Please choose between 'Yes', 'No' and 'No, and never ask me again'",
```

Comment:
Very minor: We don't provide a `pattern` property, so we won't ever see this `patternErrorMessage`! (Same with the above `codeQL.createQuery.folder` setting, though I didn't notice that one at the time 🙈)

(I can't link to the exact section, but for context, search for `patternErrorMessage` on https://code.visualstudio.com/api/references/contribution-points#Configuration-property-schema 🧩 )

Reply:
Thanks for the docs and the explanations! I've updated this to use Robert's recommended structure. Let me know if it makes sense.